### PR TITLE
[Feature] Support TRN

### PR DESCRIPTION
**Pull request by congee524**

### CODE REVIEW

File: mmaction/core/optimizer/trn_optimizer_constructor.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import torch
+from mmcv.runner import OPTIMIZER_BUILDERS, DefaultOptimizerConstructor
+from mmcv.utils import SyncBatchNorm, _BatchNorm, _ConvNd
+
+
+@OPTIMIZER_BUILDERS.register_module()
+class TSMOptimizerConstructor(DefaultOptimizerConstructor):
```
Comment (kennymckormick) [inline mmaction/core/optimizer/trn_optimizer_constructor.py:7]: Guess it should be TRNOptimizerConstructor?

**Review by kennymckormick — COMMENTED** [review]
It seems that you want to use a custom optimizer for TRN, but forget to add it?

File: mmaction/core/optimizer/trn_optimizer_constructor.py (hunk 1 above)
Comment (congee524) [reply mmaction/core/optimizer/trn_optimizer_constructor.py:7]: emmm, I forgot to delete the file. After I copied tsm_optimizer_consturtor.py and were going to modify it, I found that I can directly use TSM's.

Comment (kennymckormick) [reply mmaction/core/optimizer/trn_optimizer_constructor.py]: That's OK. So does TRN need a custom optimizer?

Comment (congee524) [reply mmaction/core/optimizer/trn_optimizer_constructor.py]: Yes :p

**Comment by dreamerlin** [comment]
Missing recognizer unittest for TRN

**Review by dreamerlin — APPROVED** [review]

**Comment by congee524** [comment]
@innerlee kindly ping :p

**Review by innerlee — APPROVED** [review]

**Comment by innerlee** [comment]
Pls rebase master :)

**Pull request merged**